pyphare/pyphare/pharesee/hierarchy/hierarchy.py

```diff
@@ -0,0 +1,742 @@
+                self.data_files.update({data_files.filename: data_files})
+            else:
+                self.data_files = {data_files.filename: data_files}
+        else:
+            self.data_files = {}
+
+        self.update()
+
+    def __getitem__(self, qty):
+        return self.__dict__[qty]
+
+    def update(self):
+        if len(self.quantities()) > 1:
+            for qty in self.quantities():
+                for time, levels in self.time_hier.items():
+                    new_lvls = {}
+                    for ilvl, level in levels.items():
+                        patches = []
+                        for patch in level.patches:
+                            patches += [Patch({qty: patch.patch_datas[qty]}, patch.id)]
+                        new_lvls[ilvl] = PatchLevel(ilvl, patches)
+                    if qty not in self.__dict__:
+                        self.__dict__[qty] = PatchHierarchy(
+                            new_lvls,
+                            self.domain_box,
+                            selection_box=self.domain_box,
+                            times=time,
+                            data_files=self.data_files,
+                        )
+                    else:
+                        self.__dict__[qty].time_hier[time] = new_lvls
+
+    def nbytes(self):
+        n = 0
+        for t in self.times():
+            for lvl in self.levels(t).values():
+                for p in lvl.patches:
+                    for pd in p.patch_datas.values():
+                        n += pd.dataset.nbytes
+        return n
+
+    def nbrPatches(self):
+        n = 0
+        for t in self.times():
+            for lvl in self.levels(t).values():
+                n += len(lvl.patches)
+        return n
+
+    @property
+    def sim(self):
+        if self._sim:
+            return self._sim
+
+        # data_files has a key/value per h5 filename.
+        # but the "serialized_simulation" in "py_attrs" should be the same for all files
+        # used by the hierarchy. So we just take the first one.
+        first_file = list(self.data_files.values())[0]
+        if "py_attrs" not in first_file.keys():
+            raise ValueError("Simulation is not available for deserialization")
+
+        from ...pharein.simulation import deserialize
+
+        try:
+            self._sim = deserialize(
+                first_file["py_attrs"].attrs["serialized_simulation"]
+            )
+        except Exception as e:
+            raise RuntimeError(f"Failed to deserialize simulation from data file : {e}")
+        return self._sim
+
+    def __call__(self, qty=None, **kwargs):
+        # take slice/slab of 1/2d array from 2/3d array
+        def cuts(c, coord):
+            return c > coord.min() and c < coord.max()
+
+        class Extractor:
+            def __init__(self):
+                self.exclusions = []
+
+            def extract(self, coord, data):
+                mask = coord == coord
```

Comment:
is this about NaNs? or is it something else, maybe `np.ones(coord.shape)` is more obvious if not about nan?